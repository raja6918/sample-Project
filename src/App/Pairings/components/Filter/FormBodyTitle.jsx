import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import ButtonMUI from '@material-ui/core/Button';
import CloseIcon from '@material-ui/icons/Close';
import ReplayIcon from '@material-ui/icons/Replay';

const FormTitle = styled.div`
  display: flex;
  justify-content: space-between;
  & h2 {
    margin-top: 0px !important;
  }
`;

const LastFilterButton = styled(ButtonMUI)`
  text-transform: none;
  font-size: 12px;
  padding: 0;
  margin-bottom: 13.28px;
  margin-right: 15px;
  &:hover {
    background-color: transparent;
  }
  & .MuiButton-startIcon {
    margin-right: 0px;
  }
`;

const ClearAllButton = styled(ButtonMUI)`
  text-transform: none;
  font-size: 12px;
  padding: 0;
  margin-bottom: 13.28px;
  &:hover {
    background-color: transparent;
  }
  & .MuiButton-startIcon {
    margin-right: 0px;
  }
`;

const FormTitleLabel = styled.h2`
  font-style: ${props => (props.isFilterChanged ? 'italic' : 'normal')};
  font-weight: bold;
  font-size: 16px;
  color: ${props => (props.isFilterChanged ? '#7e7e7e' : '#000000')} !important;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  width: 50%;
`;

const FormBodyTitle = ({
  t,
  type,
  formTitle,
  enableClearAll,
  handleClearAll,
  enableLastFilter,
  handleLastFilterApply,
  isFilterChanged,
}) => (
  <FormTitle>
    {!formTitle && <h2>{t(`FILTER.${type}`)}</h2>}
    {formTitle && (
      <FormTitleLabel isFilterChanged={isFilterChanged}>
        {formTitle}
      </FormTitleLabel>
    )}
    <div>
      <LastFilterButton
        disabled={!enableLastFilter}
        color="primary"
        startIcon={<ReplayIcon />}
        disableRipple={true}
        disableFocusRipple={true}
        onClick={handleLastFilterApply}
      >
        {t('FILTER.pane.buttons.lastFilter')}
      </LastFilterButton>
      <ClearAllButton
        disabled={!enableClearAll}
        color="primary"
        startIcon={<CloseIcon />}
        disableRipple={true}
        disableFocusRipple={true}
        onClick={handleClearAll}
      >
        {t('FILTER.pane.buttons.clearAll')}
      </ClearAllButton>
    </div>
  </FormTitle>
);

FormBodyTitle.propTypes = {
  t: PropTypes.func.isRequired,
  type: PropTypes.string,
  enableClearAll: PropTypes.bool.isRequired,
  handleClearAll: PropTypes.func.isRequired,
  enableLastFilter: PropTypes.bool,
  handleLastFilterApply: PropTypes.func.isRequired,
  formTitle: PropTypes.string.isRequired,
  isFilterChanged: PropTypes.bool,
};

FormBodyTitle.defaultProps = {
  type: 'new',
  enableLastFilter: false,
  isFilterChanged: false,
};

export default FormBodyTitle;
