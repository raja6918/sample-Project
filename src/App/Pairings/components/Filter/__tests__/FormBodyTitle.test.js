import React from 'react';
import { shallow } from 'enzyme';
import FormBodyTitle from '../FormBodyTitle';

describe('Test FormBodyTitle Component', () => {
  const minProps = {
    t: jest.fn(),
    type: 'new',
    enableClearAll: true,
    handleClearAll: jest.fn(),
  };

  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<FormBodyTitle {...minProps} />);
  });

  test('FormBodyTitle Component renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('handleClearAll should be called on clear icon click', () => {
    wrapper.find('FormBodyTitle__ClearAllButton').simulate('click');
    expect(minProps.handleClearAll).toBeCalled();
  });
});
