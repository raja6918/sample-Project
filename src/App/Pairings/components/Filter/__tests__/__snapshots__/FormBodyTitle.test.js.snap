// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FormBodyTitle Component FormBodyTitle Component renders correctly 1`] = `
<FormBodyTitle__FormTitle>
  <h2 />
  <div>
    <FormBodyTitle__LastFilterButton
      color="primary"
      disableFocusRipple={true}
      disableRipple={true}
      disabled={true}
      startIcon={<Memo />}
    />
    <FormBodyTitle__ClearAllButton
      color="primary"
      disableFocusRipple={true}
      disableRipple={true}
      disabled={false}
      onClick={[MockFunction]}
      startIcon={<Memo />}
    />
  </div>
</FormBodyTitle__FormTitle>
`;
